File: dev/python/src/routers/frame.py
from fastapi import APIRouter
from utils.setup import main_app
from utils.types import Pin, PinAndDistance

router = APIRouter(
    prefix="/frame",
    tags=["frame"],
    responses={404: {"description": "Not found"}},
)


@router.get("/")
async def read_test():
    return {"message": "here is frame!"}


@router.post("/get_frame")
async def get_frame():
    frame_base64 = main_app.get_frame()
    return {"frame_base64": frame_base64}


@router.get("/get_wrap")
async def get_wrap():
    wrap_base64, height, width = main_app.get_wrap_code()
    return {"wrap_base64": wrap_base64, "height": height, "width": width}


@router.post("/floor_setting")
async def floor_setting(data: PinAndDistance):
    print(data)
    warped_base64 = main_app.set_floor(data)
    return {"warped_base64": warped_base64}


@router.post("/zone_setting")
async def zone_setting(data: Pin):
    print(data)
    warped_with_zone = main_app.set_zone(data)
    return {"warped_with_zone": warped_with_zone}

File: dev/mobile/api/Frame.ts
import axios from "axios";

interface pinAndDistance {
    p1_p2: number;
    p1_p3: number;
    p1_p4: number;
    p2_p3: number;
    p2_p4: number;
    p3_p4: number;
    pin_1_x: number;
    pin_1_y: number;
    pin_2_x: number;
    pin_2_y: number;
    pin_3_x: number;
    pin_3_y: number;
    pin_4_x: number;
    pin_4_y: number;
}

interface pin {
    pin_1_x: number;
    pin_1_y: number;
    pin_2_x: number;
    pin_2_y: number;
    pin_3_x: number;
    pin_3_y: number;
    pin_4_x: number;
    pin_4_y: number;
}

export const fetchFrame = async (): Promise<{ frame_base64: string }> => {
    // カメラ映像を取得する
    try {
        const response = await axios.post("http://localhost:8000/frame/get_frame");
        return response.data;
    } catch (error) {
        console.error("Error:", error);
        throw new Error("データ取得に失敗しました");
    }
}

export const fetchWrap = async (): Promise<{ wrap_base64: string, height: number, width: number }> => {
    // 床エリアの映像を取得する
    try {
        const response = await axios.get("http://localhost:8000/frame/get_wrap");
        return response.data;
    } catch (error) {
        console.error("Error:", error);
        throw new Error("データ取得に失敗しました");
    }
}

export const postPinAndDistance = async (pinAndDistance: pinAndDistance): Promise<{ warped_base64: string }> => {
    try {
        const response = await axios.post(
            "http://localhost:8000/frame/floor_setting",
            pinAndDistance
        );
        return response.data;
    } catch (error) {
        console.error("Error:", error);
        throw new Error("データ取得に失敗しました");
    }
}

export const postPin = async (pin: pin): Promise<{ warped_with_zone: string }> => {
    try {
        const response = await axios.post(
            "http://localhost:8000/frame/zone_setting",
            pin
        );
        return response.data;
    } catch (error) {
        console.error("Error:", error);
        throw new Error("データ取得に失敗しました");
    }
}